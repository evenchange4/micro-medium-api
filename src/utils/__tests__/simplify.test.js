const simplify = require('../simplify');
const posts = require('./posts.json');

it('should return simplify list', () => {
  expect(simplify('username')(posts)).toMatchSnapshot();
});
