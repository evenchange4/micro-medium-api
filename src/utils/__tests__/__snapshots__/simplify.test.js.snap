// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return simplify list 1`] = `
Array [
  Object {
    "firstPublishedAt": 1501225933707,
    "subtitle": "減少 Monorepo CI 所需花費時間",
    "title": "使用 CircleCI 2.0 Workflows 挑戰三倍速",
    "url": "https://medium.com/@username/使用-circleci-2-0-workflows-挑戰三倍速-9691e54b0ef0",
  },
  Object {
    "firstPublishedAt": 1499912011732,
    "subtitle": "使用 Redux-Cylces 實行錯誤集中處理",
    "title": "Centralized Error Handle in React with RxJS",
    "url": "https://medium.com/@username/centralized-error-handle-in-react-with-rxjs-fac9e9e37e44",
  },
  Object {
    "firstPublishedAt": 1499671540298,
    "subtitle": "Landing Page for MCS Lite",
    "title": "React Stack 開發體驗與優化策略",
    "url": "https://medium.com/@username/react-stack-開發體驗與優化策略-b056da2fa0aa",
  },
  Object {
    "firstPublishedAt": 1495693600421,
    "subtitle": "Lazy Load React Component with React-Loadable",
    "title": "Component-based Code Splitting",
    "url": "https://medium.com/@username/component-based-code-splitting-70aeebbb2dc7",
  },
  Object {
    "firstPublishedAt": 1494422818494,
    "subtitle": "以一個簡單到不行的大小寫為例，不單只是 Coding Style 或命名規範，你可能會需要 Eslint 來幫你盡量降低這種小意外的發生。",
    "title": "這個雷踩了四次，為什麼你可能需要 Eslint",
    "url": "https://medium.com/@username/這個雷踩了四次-為什麼你可能需要-eslint-9330c1245ec1",
  },
  Object {
    "firstPublishedAt": 1493908671942,
    "subtitle": "Recently, I have read an article by José M. Pérez about How Medium does progressive image loading, and that is what I need about the lazy…",
    "title": "Reproducing Medium Style Progressive Image Loading for React",
    "url": "https://medium.com/@username/reproducing-medium-style-progressive-image-loading-for-react-2e83bba0c608",
  },
  Object {
    "firstPublishedAt": 1493193304597,
    "subtitle": "MCS Lite’s Front-End Tech Stack for Mobile",
    "title": "Build A Web App in MediaTek",
    "url": "https://medium.com/@username/build-a-web-app-in-mediatek-61b0a26215a0",
  },
  Object {
    "firstPublishedAt": 1493083372209,
    "subtitle": "開發板の隨手筆記 - 封裝 MCS API for Arduino",
    "title": "Writing a Library for LinkIt 7697",
    "url": "https://medium.com/@username/writing-a-library-for-linkit-7697-9e562d28e17",
  },
  Object {
    "firstPublishedAt": 1492745328047,
    "subtitle": "多國語言的機制建議盡量在專案初期就能有好的規劃，不至中後期要把字串抽出將會是件累人的事，而在現今前端開發的演進下，多國語言的流程也出現了一些不同的處理方式，以下簡介我們在 MediaTek 內前端專案處理的模式。",
    "title": "I18n Workflow for React Project",
    "url": "https://medium.com/@username/i18n-workflow-for-react-project-9f9ff8fe9aef",
  },
  Object {
    "firstPublishedAt": 1492590593069,
    "subtitle": "Prettier v1.0.0 就在四月初已經正式釋出，等同宣稱在 Production 環境也可安心穩定地使用了，也是時候考慮導入至專案中囉。在專案的導入順序我想應該會是 Prettier 再來才是 Eslint 的設定，兩者是可以並存的，以下以我覺得的導入順序做介紹。",
    "title": "Introducing Prettier with Eslint",
    "url": "https://medium.com/@username/introducing-prettier-with-eslint-13f567ae0184",
  },
]
`;
